examples/sighup-example.rs

```diff
@@ -0,0 +1,33 @@
+extern crate futures;
+extern crate tokio_core;
+extern crate tokio_signal;
+
+use futures::stream::Stream;
+use tokio_core::reactor::Core;
+use tokio_signal::unix::{Signal,SIGHUP};
+
+fn main() {
+    // set up a Tokio event loop
+    let mut core = Core::new().unwrap();
+
+    // on Unix, we can listen to whatever signal we want, in this case: SIGHUP
+    let sighup = Signal::new(SIGHUP, &core.handle());
+    let stream = core.run(sighup).unwrap();
```

Comment:
I think I should be able to do without this `core.run()`, but any attempt to use `.flatten_stream()` immediately devolves into compiler errors saying that some extremely verbose trait-bound is not satisfied.
This exceeds my current skill / wakefulness level (slightly past adult bed-time here)